### PR TITLE
Added the updated draft of Hello NPU Notebook

### PR DESCRIPTION
**Pull request by ujjayant-kadian**
Adding a draft of Hello NPU Notebook, that will be released alongside OV24.1

### CODE REVIEW

**Comment by mlyashko** [comment]
@eaidova , could you please take a look? This is for 2024.1 - introducing NPU as a device in dedicated notebook.

File: notebooks/hello-npu/hello-npu.ipynb [hunk 1]
```diff
@@ -0,0 +1,1253 @@
+{
```
Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    %pip install -q "openvino-dev&gt;=2024.1.0"
</pre><p>list of dependencies looks strange... in code below, torch and torchvision used for mode conversion, it does not looks like tensorflow is required to run this notebook.</p><p>Also we avoid to use openvino-dev as it contains deprecated tools (model_optimizer), but code bellow used benchmark app and convert_model that are part of openvino package.</p><p> </p><p>Also openvino 2024.1 is not released, but I believe, it is possible to launch this notebook using openvino-nightly, right?</p><p>Please use openvino-nightly for unblocking notebook merge before release package will be published </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>I think description in the brackets is not applicable for notebook, as it includes installation via pypi inside. It will be hard to maintain and track these changes apilcability from release to release and we can forgot to update it when other distribution channels will be also available</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>please provide link on resnet50 model</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    ONNX_CV_MODEL_PATH = MODEL_DIRECTORY_PATH / f'{model_name}.onnx'
</pre><p>does it really required to convert model to onnx? is it not possible to use pytorch frontend directly? (if you use onnx only for avoiding dynamic shapes, you can pass input parameter with required shape into ov.convert_model.</p><p>Please check <a href="https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/pytorch-to-openvino/pytorch-to-openvino.ipynb" target="_blank">https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/pytorch-to-openvino/pytorch-to-openvino.ipynb</a></p><p>for more examples</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>I think it maybe difficult to find lines with these changes in huge benchmark_app log. Can we highlight somehow them (provide info on which lines required to pay attention or maybe parse longs and provide simplified longs)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='63'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>which one googlenet-v4? Could you please provide link?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='70'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p> </p><p>I think this page reference may give additional educational value, could you please integrate link into intro?</p><p><a href="https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.html" target="_blank">https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.html</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>I think it is also important to note that NPU also support INT8 quantized models here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>I think is obvious summarization which does not bring new information for user. Maybe better to link some additional resources demos, notebooks that can be launched using NPU in this section?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='72'/>

**Comment by eaidova** [comment]
@ujjayant-kadian  our github actions ci unfortunatly does not have NPU power, so it is impossible to check notebook in precommit, please add it into ignored lists:
https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_treon_linux.txt
https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_treon_win.txt
https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_treon_mac.txt
https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_treon_docker.txt
https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_convert_execution.txt

**Comment by eaidova** [comment]
Also please add unknown words detected here https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/8680581743/job/23801469001?pr=1898 for our spell checker tool into https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/spellcheck/.pyspelling.wordlist.txt

I'm not sure about AlexNet only, possibly it should be replaced with resnet50? (it may looks confusing for users who checking resnet50 result inside notebook read about irrelevant AlexNet accuracy is not great and recommendation to use GoogleNet while in notebook used resnet50 that is also solving the same classification task, it may raise some concerns and untrusting to provided notebook and as it gives some nonactual recommendations to use very old models...)

File: notebooks/hello-npu/hello-npu.ipynb (hunk 1 above)
Comment (ujjayant-kadian) [reply notebooks/hello-npu/hello-npu.ipynb:1]: I see, I just picked up this information from [OpenVINO documentation](https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.html#limitations). I think I will remove this line as it doesn't say much and I am using resnet50 anyways for classification tasks.

Comment (ujjayant-kadian) [reply notebooks/hello-npu/hello-npu.ipynb:1]: Alright, makes sense. Changed!

Comment (ujjayant-kadian) [reply notebooks/hello-npu/hello-npu.ipynb:1]: I have provided the link on resnet50 model. I have attached the paper as well. What particular link are you talking about here?

Comment (eaidova) [reply notebooks/hello-npu/hello-npu.ipynb:1]: <p>I speaking about link on pretrained model source <a href="https://pytorch.org/vision/main/models/generated/torchvision.models.resnet50.html" target="_blank">https://pytorch.org/vision/main/models/generated/torchvision.models.resnet50.html</a></p>

File: notebooks/hello-npu/hello-npu.ipynb [hunk 2]
```diff
@@ -0,0 +1,1249 @@
...
```
Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p> </p><p>I think this part is more related to general openvino API and can be removed:</p><p><s>Any of the device names returned by the </s><code><s>available_devices</s></code><s> method are valid device specifiers.   </s></p><p><s>You may also use "AUTO", which will automatically select the best device for inference.     To learn more about AUTO plugin, visit the </s><a href="https://docs.openvino.ai/2024/openvino-workflow/running-inference/inference-devices-and-modes/auto-device-selection.html" target="_blank"><s>Automatic Device Selection</s></a><s> page.</s></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (eaidova) [inline notebooks/hello-npu/hello-npu.ipynb:1]: <p>could you please add --extra-index-url <a href="https://download.pytorch.org/whl/cpu" target="_blank">https://download.pytorch.org/whl/cpu</a></p><p> </p><p>this is needed for avoid downloading torch with cuda dependency (saves a lot of  disk space) if torch is not installed</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

**Review by eaidova — APPROVED** [review]

**Comment by ujjayant-kadian** [comment]
@eaidova I have implemented those final changes. Thank you!
